### ISSUE TITLE
Sync: security review 

### ISSUE THREAD

**Issue by arkpar** [paritytech/polkadot-sdk#569]
Syncing code should be reviewed for any potential

* [ ] Eclipse or attention stealing attacks.
* [ ] DOS attacks on memory. All data collected per peer or globally should be bounded.
* [ ] DOS attacks on the CPU. There should not be potentially slow request, and in case there are, there should be a throttling mechanism.

**Comment by arkpar** [comment]
This issue is outdated. @dmitry-markin Could you please check if any of this is still relevant?

**Comment by dmitry-markin** [comment]
IMO it's not relevant:

1. _Eclipse or attention stealing attacks_ should not be an issue, because we request multiple peers in parallel and peers are selected randomly from the ones discovered in the DHT.
2. _DOS attacks on memory_ — the maximum number of blocks per request is bounded as well as the response size in the request-response protocol. One thing that comes to mind is [`BlockCollection`](https://github.com/paritytech/polkadot-sdk/blob/3dfbdf4a454f35238500779e503e1ec32ba7fc63/substrate/client/network/sync/src/blocks.rs#L58) — out of my head I can't tell if it's possible to deliberately form gaps in the block range to hold off importing the entire range and consume memory. Probably not likely, because the canonical chain will be downloaded from other peers as well, and if it's actually a fork, a reorg will eventually happen. @arkpar can you double check my reasoning here?
3. _DOS attacks on the CPU_ — we are bounded by the DB here. There is no timeout in the block request handler, nor there is a generic timeout for processing incoming requests.

**Comment by dmitry-markin** [comment]
Regarding the unbounded channels — litep2p uses bounded channels for incoming requests, but import queue still uses `tracing_unbounded` channel.

**Comment by arkpar** [comment]
> DOS attacks on memory — the maximum number of blocks per request is bounded as well as the response size in the request-response protocol. One thing that comes to mind is [BlockCollection](https://github.com/paritytech/polkadot-sdk/blob/3dfbdf4a454f35238500779e503e1ec32ba7fc63/substrate/client/network/sync/src/blocks.rs#L58) — out of my head I can't tell if it's possible to deliberately form gaps in the block range to hold off importing the entire range and consume memory. Probably not likely, because the canonical chain will be downloaded from other peers as well, and if it's actually a fork, a reorg will eventually happen. @arkpar can you double check my reasoning here?

Sounds correct.

> Regarding the unbounded channels — litep2p uses bounded channels for incoming requests, but import queue still uses tracing_unbounded channel.

The import queue has an internal limit on the number of blocks that can be queued. So I imagine this is not much of an problem? If it is, could you please file a separate issue

**Comment by dmitry-markin** [comment]
> The import queue has an internal limit on the number of blocks that can be queued. So I imagine this is not much of an problem? If it is, could you please file a separate issue

I don't think there is such a limit in the import queue. Here is a separate issue for implementing a backpressure from the import queue to the networking layer: https://github.com/paritytech/polkadot-sdk/issues/9607

**Comment by arkpar** [comment]
Ok, closing this then
